### ISSUE TITLE
ECMAScript 6 and Block Scope

### ISSUE THREAD

**Issue by xPetrus** [airbnb/javascript#406]
Add block scope to let & const bullet under ECMAScript 6 Styles section or give it its own overview and link.

**Comment by nischayv** [comment]
Seems like this is already covered in [2.3](https://github.com/airbnb/javascript#references--block-scope). Am I wrong?

**Comment by ljharb** [comment]
Somewhat, but this issue is looking for a more in-depth section to be added to https://github.com/airbnb/javascript#ecmascript-6-es-2015-styles

**Comment by nischayv** [comment]
I was thinking of adding additional examples like using let in loops under 2.3 itself and referencing it in the es6 list. Does that sound good?

**Comment by ljharb** [comment]
I guess that could work, but I don't feel like "references" is really the best place for the actual meat of the content.

**Comment by nischayv** [comment]
That's true. But it seemed to somewhat cover a few concepts already. But I can create a new section entirely for block scope and have stuff like shadowing variables, loops, etc.  Would that work?

**Comment by ljharb** [comment]
Hmm - that might be better, but I'm hesitant to rashly do a big reorg of parts of the guide. If you don't mind the possibly thrown away work, I'd love to review a PR - otherwise I'll think on it more first.

**Comment by nischayv** [comment]
Sounds good!
